CRM/Utils/String.php

```diff
@@ -362,7 +362,7 @@ public static function match($url1, $url2) {
     $url2Str = parse_url($url2);
 
     if ($url1Str['path'] == $url2Str['path'] &&
-      self::extractURLVarValue(CRM_Utils_Array::value('query', $url1Str)) == self::extractURLVarValue(CRM_Utils_Array::value('query', $url2Str))
+      self::extractURLVarValue($url1Str['query'] ?? '') == self::extractURLVarValue($url2Str['query'] ?? '')
```

Comment:
Code suggest doesn't want to let me suggest this - not sure why.  I know you didn't change this, but naming something `$url1Str` suggests it is a string - which it isn't. So maybe change all `$url1Str` to `$url1Components` and `$url2Str` to `$url2Components` ?

Reply:
Ok @aydun I've cleaned up that function and given the variables less-confusing names.